src/python_testing/TC_DISHALM_3_1.py

```diff
@@ -0,0 +1,139 @@
+
+    async def read_event_notify_state(self, endpoint):
+        return await self.default_controller.ReadEvent(self.dut_node_id, [(endpoint, Clusters.DishwasherAlarm.Events.Notify.state)])
+
+    async def read_event_notify(self, endpoint):
+        return await self.default_controller.ReadEvent(self.dut_node_id, [(endpoint, Clusters.DishwasherAlarm.Events.Notify)])
+
+    async def read_state_attribute(self, endpoint):
+        return await self.read_da_attribute_expect_success(endpoint, Clusters.DishwasherAlarm.Attributes.State)
+
+    async def read_latch_attribute(self, endpoint):
+        return await self.read_da_attribute_expect_success(endpoint, Clusters.DishwasherAlarm.Attributes.Latch)
+
+    async def send_reset_cmd(self, setAlarm) -> Clusters.Objects.DishwasherAlarm.Commands.Reset:
+        ret = await self.send_single_cmd(cmd=Clusters.Objects.DishwasherAlarm.Commands.Reset.alarms(setAlarm=setAlarm), endpoint=self.endpoint)
+        return ret
+
+    @async_test_body
+    async def test_TC_DISHALM_3_1(self):
+
+        endpoint = self.user_params.get("endpoint", 1)
+
+        asserts.assert_true(self.check_pics("DISHALM.S.A0001"), "DISHALM.S.A0001 must be supported")
+        asserts.assert_true(self.check_pics("DISHALM.S.A0002"), "DISHALM.S.A0002 must be supported")
+
+        self.print_step(1, "1a. Commissioning, already done")
+
+        self.print_step(2, "1b. Set up subscription to Notify Event")
+        await self.subscribe_event(endpoint=endpoint)
+
+        self.print_step(3, "2a. Operate Device to set the condition to raise the Inflow alarm and wait a few seconds")
+        input("Press Enter when done.\n")
+
+        notify = self.read_event_notify(endpoint=endpoint)
+        notify_state = self.read_event_notify_state(endpoint=endpoint)
+
+        logging.info("Notify: %s" % (notify))
+        logging.info("Notify State: %s" % (notify_state))
+
+        asserts.assert_true(notify_state & Clusters.DishwasherAlarm.Bitmaps.AlarmMap.kInflowError, "State bit 0 is not set to TRUE")
+
+        self.print_step(4, "2b. Read from the DUT the State Attribute")
+        state = self.read_state_attribute(endpoint=endpoint)
+
+        logging.info("State: %s" % (state))
+
+        asserts.assert_true(state & Clusters.DishwasherAlarm.Bitmaps.AlarmMap.kInflowError, "Bit 0 of State is not set to 1")
+
+        self.print_step(5, "2c. Operate Device to set the condition to lower the Inflow alarm and wait a few seconds")
+        input("Press Enter when done.\n")
+
+        notify = self.read_event_notify(endpoint=endpoint)
+        notify_state = self.read_event_notify_state(endpoint=endpoint)
+
+        logging.info("Notify: %s" % (notify))
+        logging.info("Notify State: %s" % (notify_state))
+
+        latch = self.read_latch_attribute(endpoint=endpoint)
+        logging.info("Latch: %s" % (latch))
+
+        if latch == 0:
+            asserts.assert_false(notify_state & Clusters.DishwasherAlarm.Bitmaps.AlarmMap.kInflowError,
```

Comment:
This check happens in `2d`, is it needed here?

Reply:
Yes, but I could clean it up a bit